algorithms/scaling/observers.py

```diff
@@ -191,9 +197,37 @@ def print_scaling_summary(self, scaling_script):
             scaling_script.params.reflection_selection.min_partiality,
         )
         if MergingStatisticsObserver().data:
+            merging_stats = MergingStatisticsObserver().data["statistics"]
+            logger.info(make_merging_statistics_summary(merging_stats))
+            vals = cc_half_fit(merging_stats, limit=0.3)
+            r_cc = vals[0]
+            max_current_res = merging_stats.bins[-1].d_min
+            cut_merging_statistics_result = None
+            cut_anom_merging_statistics_result = None
+            if r_cc - max_current_res > 0.005:
+                logger.info(
+                    "Resolution limit suggested from cc1/2 fit (limit cc1/2=0.3): %.2f",
+                    r_cc,
+                )
+                try:
+                    (
+                        cut_merging_statistics_result,
+                        cut_anom_merging_statistics_result,
+                    ) = merging_stats_from_scaled_array(
+                        scaling_script.scaled_miller_array.resolution_filter(
+                            d_min=r_cc
+                        ),
+                        scaling_script.params.output.merging.nbins,
+                        scaling_script.params.output.use_internal_variance,
+                    )
```

Comment:
Surely this screams at some kind of need for a named tuple or something? The line / variable lengths are absurd here 🙂